src/lib/result.ts

```diff
@@ -0,0 +1,390 @@
+import { Principal } from '@dfinity/principal';
```

Comment:
I feel like `result.ts` is bad name for this file. Maybe `resultHandler` or `displayResult` would be better.

Reply:
I have chosen resultHandler